lib/screens/authentication/import_key/interactor/mappers/import_key_state_mapper.dart

```diff
@@ -6,22 +6,30 @@ import 'package:seeds/screens/authentication/import_key/import_key_errors.dart';
 import 'package:seeds/screens/authentication/import_key/interactor/viewmodels/import_key_bloc.dart';
 
 class ImportKeyStateMapper extends StateMapper {
-  ImportKeyState mapResultsToState(ImportKeyState currentState, List<Result> results, AuthDataModel authData) {
+  ImportKeyState mapResultsToState({
+    required ImportKeyState currentState,
+    required AuthDataModel authData,
+    required List<Result> results,
+    AuthDataModel? alternateAuthData,
+    List<Result> alternateResults = const [],
+  }) {
+    final List<Result> goodResults = results.isNotEmpty ? results : alternateResults;
+    final AuthDataModel? goodAuthData = results.isNotEmpty ? authData : alternateAuthData;
     // No account found. Show error
-    if (results.isEmpty) {
+    if (goodResults.isEmpty) {
       return currentState.copyWith(pageState: PageState.failure, error: ImportKeyError.NoAccountsFound);
     }
 
     // Accounts found, but errors fetching data happened.
-    if (areAllResultsError(results)) {
+    if (areAllResultsError(goodResults)) {
```

Comment:
So is it in these 2 error checks where you want to repeat the search account again?